include/BSMPT/utility/utility.h

```diff
@@ -46,7 +46,48 @@ bool StringEndsWith(const std::string &str, const std::string &suffix);
 const std::string sep = "\t";
 
 /**
- * Overload to print out vectors with the << operator
+ * @brief factorial function
+ */
+int factorial(const int &a);
+
+/**
+ * @brief push back vector into vector
+ */
+template <typename T>
+std::vector<T> push_back(std::vector<T> &a, const std::vector<T> &b)
+{
+  return a.insert(a.end(), b.begin(), b.end());
+}
+
+/**
+ * @brief vector to_string
+ */
+template <typename T> std::string vec_to_string(const std::vector<T> &vec)
+{
+  std::string res;
+  bool first = true;
+  for (const auto &el : vec)
+  {
+    if (not first)
+    {
+      res += " " + std::to_string(el);
```

Comment:
```suggestion
      res += sep + std::to_string(el);
```

We use the sep variable for all the output files, why not use it here?

Reply:
sep = "\t" and that is a space. Our outputs are .tsv so we need to use a single space there.

Reply:
I don't get why we need a single space here. Is this printed in a column of the output file and then leads to a problem?

Reply:
You are correct, it is not so we can change it.